app/src/org/commcare/activities/FormRecordListActivity.java

```diff
@@ -397,8 +397,12 @@ public void onCreateContextMenu(ContextMenu menu, View v, ContextMenuInfo menuIn
             menu.add(Menu.NONE, DELETE_RECORD, DELETE_RECORD, Localization.get("app.workflow.forms.delete"));
         }
 
-        if (FormRecord.STATUS_QUARANTINED.equals(value.getStatus())) {
-            menu.add(Menu.NONE, RESTORE_RECORD, RESTORE_RECORD, Localization.get("app.workflow.forms.restore"));
+        if (FormRecord.STATUS_QUARANTINED.equals(value.getStatus()) &&
+                !FormRecord.QuarantineReason_LOCAL_PROCESSING_ERROR.equals(value.getReasonForQuarantine())) {
```

Comment:
It seems a bit weird that we're comparing what are essentially two human-readable error messages for equality here (rather than using, say, an Enum or String key). From what I can tell the semantic meaning of these strings is never used (IE the error messages saved under "reason" are never displayed to the user). Is there any reason not to turn these into an enum or simple keys?

Reply:
Ha yea fair point, so this looks a little weirder than it actually was in my mind because I implemented this PR with an eye to the fact that https://github.com/dimagi/commcare-android/pull/1721 will hopefully get merged sometime soon-ish (waiting on HQ work with no clear timeline). But I essentially copied over a bit of what was already in that PR to here, and there are further changes in that PR that would actually be using this string as something to display to the user. 

As far as why not to make this an enum, my reason was that the `reasonForQuarantine` field is set as a `@Persisting` value in `FormRecord`, which I believe means it has to be a primitive (although I could be wrong about that?). So once I was making these variables Strings anyway, I figured it didn't actually matter whether the string values were short strings or long strings, and so they could double as identifiers and messages. 

Let me know if you still think this is a bad idea given those explanations.

Reply:
Generally agreed that it doesn't seem quite right for this to serve double-duty, and it also has the disadvantage that if we change the error message it will (quite unexpectedly) result in the forms being restorable.

One alternative is to simply store the "status before quarantine", rather than flagging as "unrecoverable" essentially, so that records which are quarantined from "unprocessed" get returned to "unprocessed" rather than "unsent"

Reply:
Fixed this